tests/k8s-azure/manifest/cluster-api/cluster-template-prow-dual-stack-md.yaml

```diff
@@ -211,6 +211,9 @@ metadata:
 spec:
   template:
     spec:
+      identity: UserAssigned
+      userAssignedIdentities:
+      - providerID: ${AZURE_USER_ASSIGNED_IDENTITY_PROVIDER_ID}
```

Comment:
@nilo19 Where do you expect this variable to get set? This isn't something CAPZ sets AFAICT. Our templates use something like this here:
```yaml
      - providerID: /subscriptions/${AZURE_SUBSCRIPTION_ID}/resourceGroups/${CI_RG}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${USER_IDENTITY}
```
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/a498485f56254673096d2184b5db1ec950240693/templates/test/ci/cluster-template-prow.yaml#L142

Reply:
that's a mistake, thanks for catching this.

Reply:
https://github.com/kubernetes-sigs/cloud-provider-azure/pull/6725 to fix this